src/Components/Components/test/Forms/EditContextDataAnnotationsExtensionsTest.cs

```diff
@@ -0,0 +1,171 @@
+            editContext.NotifyFieldChanged(requiredStringIdentifier);
+            Assert.Equal(1, onValidationStateChangedCount);
+            Assert.Equal(new[] { "The RequiredString field is required." }, editContext.GetValidationMessages());
+
+            // Act/Assert 2: Fix RequiredString, but only notify about IntFrom1To100
+            // Only IntFrom1To100 gets validated; messages for RequiredString are left unchanged
+            model.RequiredString = "This string is very cool and very legal";
+            editContext.NotifyFieldChanged(intFrom1To100Identifier);
+            Assert.Equal(2, onValidationStateChangedCount);
+            Assert.Equal(new string[]
+                {
+                    "The RequiredString field is required.",
+                    "The field IntFrom1To100 must be between 1 and 100."
+                },
+                editContext.GetValidationMessages());
+
+            // Act/Assert 3: Notify about RequiredString
+            editContext.NotifyFieldChanged(requiredStringIdentifier);
+            Assert.Equal(3, onValidationStateChangedCount);
+            Assert.Equal(new[] { "The field IntFrom1To100 must be between 1 and 100." }, editContext.GetValidationMessages());
+        }
+
+        [Theory]
+        [InlineData(nameof(TestModel.ThisWillNotBeValidatedBecauseItIsAField))]
+        [InlineData(nameof(TestModel.ThisWillNotBeValidatedBecauseItIsInternal))]
+        [InlineData("ThisWillNotBeValidatedBecauseItIsPrivate")]
+        [InlineData("This does not correspond to anything")]
+        [InlineData("")]
+        public void IgnoresFieldChangesThatDoNotCorrespondToAValidatableProperty(string fieldName)
+        {
+            // Arrange
+            var editContext = new EditContext(new TestModel()).AddDataAnnotationsValidation();
+            var onValidationStateChangedCount = 0;
+            editContext.OnValidationStateChanged += (sender, eventArgs) => onValidationStateChangedCount++;
+
+            // Act/Assert: Ignores field changes that don't correspond to a validatable property
+            editContext.NotifyFieldChanged(editContext.Field(fieldName));
+            Assert.Equal(0, onValidationStateChangedCount);
+
+            // Act/Assert: For sanity, observe that we would have validated if it was a validatable property
+            editContext.NotifyFieldChanged(editContext.Field(nameof(TestModel.RequiredString)));
+            Assert.Equal(1, onValidationStateChangedCount);
+        }
+
+        class TestModel
+        {
+            [Required] public string RequiredString { get; set; }
+
+            [Range(1, 100)] public int IntFrom1To100 { get; set; }
+
+#pragma warning disable 649
+            [Required] public string ThisWillNotBeValidatedBecauseItIsAField;
+            [Required] string ThisWillNotBeValidatedBecauseItIsPrivate { get; set; }
+            [Required] internal string ThisWillNotBeValidatedBecauseItIsInternal { get; set; }
```

Comment:
BTW putting data annotations messages in tests is kinda a pit of failure. You have to do work to make sure the tests only run in a single locale, and we can also break if a different platform defines the messages differently.

Reply:
Yeah, I felt a bit bad about doing it the first time.

Anyway now I've put in explicit messages for all the rules we assert about, so it's independent of locale/platform.